script/build/README.md

```diff
@@ -0,0 +1,110 @@
+# Simplified build system
+
+This directory contains the simplified build system. You can use this to
+compile, and cross compile, Measurement Kit and its dependencies. Specifically
+the following are the _packages_ that you can compile:
+
+- `geoip`: [geoip](https://github.com/maxmind/geoip-api-c)
+- `libressl`: [libressl](https://github.com/libressl-portable)
+- `libevent`: [libevent](https://github.com/libevent/libevent)
+- `c-ares`: [c-ares](https://github.com/c-ares/c-ares)
+- `mk`: [MK](https://github.com/measurement-kit/measurement-kit) itself
+
+If needed, the build system will fetch the required sources and possibly
+apply patches. In such case, sources will be downloaded in
+
+```
+./MK_BUILD/<os>/<arch>/<package>/<package-sources-dir>/
+```
+
+Artifacts (i.e. libraries and headers) will be installed in
+
+```
+./MK_DIST/<os>/<arch>/<package>/
+```
+
+(Paths are relative to the top level directory of MK sources.)
+
+If a package needs another package, you need to compile that other package
+first (or to populate appropriately `./MK_DIST/<os>/<arch>/<package>/` using
+the artifacts generated by a previous build).
+
+All the scripts except `build.bat` can actually be called from everywhere.
+
+## Build on Linux / macOS
+
+Just call the script named after the package to build. E.g.:
+
+```
+./script/build/libressl
+```
+
+In addition to Unix essential commands, you'll need:
+
+- make
+- autoconf and automake
+- g++ or clang (must be C++14-capable)
+- libstdc++ or libc++ (again must be C++14-capable)
+- git
+- libtool
+- wget
+- gpg
+
+## Cross compile for iOS
+
+We assume that you have a macOS system with Xcode installed and the command
+line developer tools installed.
+
+```
+./script/build/cross-ios <os> <arch> ./script/build/<package>
+```
+
+To see available `<os>` and `<arch>` values, run `cross-ios` without arguments.
+
+## Cross compile for Android
+
+We assume that you have the Native Development Kit (NDK) installed. The simplest
+and recommended way to get the NDK is to install it along with Android Studio.
+
+Create a standalone Android toolchain using the `android-toolchain` script:
+
+```
+./script/build/android-toolchain <ndk-bundle-path> <arch>
+```
+
+Run the script without arguments to see all the available `<arch>` options.
+The toolchain will be created in:
+
+```
+./MK_BUILD/android/toolchain/<arch>/
+```
+
+Then cross compile using:
+
+```
+./script/build/cross-android <arch> ./script/build/<package>
+```
+
+## Compile for Windows
+
+We assume that:
+
+1. you have installed Visual Studio 2017 Community
+
+2. you have installed CMake and opted to have `cmake.exe` inside your path
+   during the install process (important, as that is not the default
+   choice)
+
+3. you have installed Ubuntu and `ubuntu.exe` is available from your
+   path (that should be the case), and you have installed into Ubuntu
+   on Windows all the Linux packages mentioned above when discussing
+   building for Linux or macOS
```

Comment:
This is a pretty strong requirements for windows. Keep in mind that this is only available since Windows 10.

I am OK with keeping it like this for the moment, but we should see if it's possible to do it what we need with standard windows tooling.

Reply:
Please: no. If we want to support _building_ on previous versions of Windows, then we would better have everything in the repository and minimise _our_ pain in maintaining.

Having `ubuntu.exe` seems to me a _good_ compromise between having to learn to develop on Windows like a pro and using tooling that we know to do the job.

Reply:
(Also, to the maximum extent as possible, I'd like to minimise the amount of work entailed by maintaining more than one build system. To this end, for example, I would like to make sure that in the future we _only_ build Measurement Kit using CMake. No more autotools.)

Reply:
Also: why should we support _building_ on a version of Windows that we don't have. Really, I do not understand why we would want doing that. Please, let's try to keep the building on Windows simple _for us_, given that we have limited resources. Also: building on Windows 10 is not impairing in any way _using_ Measurement Kit in earlier Windows versions.

Reply:
Sounds reasonable.